app/custom-my-store/custom-prod-refer/custom-prod-refer.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit } from '@angular/core';
+import { ActiveCartService, Product, ProductService } from '@spartacus/core';
+import { CurrentProductService } from '@spartacus/storefront';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-custom-prod-refer',
+  templateUrl: './custom-prod-refer.component.html',
+  styleUrls: ['./custom-prod-refer.component.scss']
+ 
+})
+export class CustomProdReferComponent implements OnInit  {
+
+  productId1: any = 1981414;
+  productId2: any = 1981412;
+  productId3: any = 1981413;
+
+  myCurrentProduct1$: Observable<Product|any> = this.productService.get(this.productId1);
```

Comment:
Hi @nitishgupta-knack, I don't know where you took the reference for this productService, this is one way to get the product details and display them. Here you have expressly declared the product IDs and called the service individually, what if you have like 10 or 20 products?

Hence this is not the right approach.